platform/qt/src/qmapboxgl.cpp

```diff
@@ -1547,14 +1555,15 @@ QMapboxGLPrivate::QMapboxGLPrivate(QMapboxGL *q, const QMapboxGLSettings &settin
         settings.assetPath().toStdString(),
         settings.cacheDatabaseMaximumSize()))
     , threadPool(mbgl::sharedThreadPool())
-    , mapObj(std::make_unique<mbgl::Map>(
-        *this, mbgl::Size{ static_cast<uint32_t>(size.width()), static_cast<uint32_t>(size.height()) },
-        pixelRatio, *fileSourceObj, *threadPool,
-        mbgl::MapMode::Continuous,
-        static_cast<mbgl::GLContextMode>(settings.contextMode()),
-        static_cast<mbgl::ConstrainMode>(settings.constrainMode()),
-        static_cast<mbgl::ViewportMode>(settings.viewportMode())))
 {
+    mapObj = std::make_unique<mbgl::Map>(
+            *this, sanitizedSize(size),
```

Comment:
Because `QSize` default constructor initializes witdh and height with -1. This causes an overflow when converting from `int` to `uint32_t`- using `std::numeric_limits<uint32_t>::max()` for width and height has proven to cause undefined behavior in places like [tileCover](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/util/tile_cover.cpp#L159-L160).

Qt sometimes initializes a dummy map plugin as means of discovering which map plugins it has available - when it does, it uses a default `QSize` for initializing the item size.